src/kernels/webgl/reduce_gpu.ts

```diff
@@ -59,6 +64,8 @@ export class ReduceProgram implements GPGPUProgram {
     let updateSnippet = `
       if (${reduceType === 'sum'}) {
         sumValue += dot(values, ones);
+      } else if (${reduceType === 'prod'}) {
+        prodValue *= values[0] * values[1] * values[2] * values[3];
```

Comment:
I happened to trace the code here recently. since the `updateSnippet` would be part of the program code, directly performing the necessary calculation without if/else condition may improve the performance, especially `updateSnippet` is inside the for loop.

so the GPGPUProgram would look like this:
```
        for (int i = 0; i < 0; i += 4) {                                        
          int inIdx = inOffset + i;                                             
          vec4 values = vec4(                                                   
            getValue(batch, inIdx),                                             
            getValue(batch, inIdx + 1),                                         
            getValue(batch, inIdx + 2),                                         
            getValue(batch, inIdx + 3)                                          
          );                                                                    
                                                                                
          prodValue *= values[0] * values[1] * values[2] * values[3];  // only the calculation but no if/else         
        } 
```

Reply:
@yhwang thanks for the suggestion, curious, have you done any measure on this improvement?